### ISSUE TITLE
Build error using Kotlin 2.0.20, Koin 4.0 and Koin Annotations when declaring shared modules in a KMP project

### ISSUE THREAD

**Issue by vinceglb** [InsertKoinIO/koin-annotations#187]
**Describe the bug**
After following Arnaud Giuliani talk in DevFest Toulouse, I tried to migrate my KMP project to koin-annotations. But some parts were without success.

The migration is working fine until it comes into declaring shared modules between native targets when using the following versions:

```toml
koin = "4.0.0"
koin-annotations = "2.0.0-Beta1"
kotlin = "2.0.20"
ksp = "2.0.20-1.0.25"
```

I tried to follow the setup from the template [InsertKoinIO/KMP-App-Template](https://github.com/InsertKoinIO/KMP-App-Template):

commonMain module: Koin.kt
```kotlin
@Module
@ComponentScan("com.jetbrains.kmpapp.data")
class DataModule

@Module
@ComponentScan("com.jetbrains.kmpapp.screens")
class ViewModelModule

@Module
expect class NativeModule()    // I needed to add "()" here, if not it throws another error

@Module(includes = [DataModule::class,ViewModelModule::class, NativeModule::class])
class AppModule
```

androidMain / iosMain module: Koin.android.kt / Koin.ios.kt
```kotlin
@Module
@ComponentScan("com.jetbrains.kmpapp.native")
actual class NativeModule
```

But when I tried to build the app, I got this error message:
```bash
e: file:///Users/vince/StudioProjects/KMP-App-Template/composeApp/build/generated/ksp/metadata/commonMain/kotlin/org/koin/ksp/generated/AppModuleGencom$jetbrains$kmpapp$di.kt:8:143 None of the following candidates is applicable:
val AppModule.module: Module
val DataModule.module: Module
val ViewModelModule.module: Module
```

Here is the generated file causing the error:
```kotlin
package org.koin.ksp.generated

import org.koin.core.module.Module
import org.koin.dsl.*


public val com_jetbrains_kmpapp_di_AppModule : Module get() = module {
	includes(com.jetbrains.kmpapp.di.DataModule().module,com.jetbrains.kmpapp.di.ViewModelModule().module,com.jetbrains.kmpapp.di.NativeModule().module)
}
public val com.jetbrains.kmpapp.di.AppModule.module : org.koin.core.module.Module get() = com_jetbrains_kmpapp_di_AppModule
```

<details><summary>Details</summary>
<p>
<img width="819" alt="image" src="https://github.com/user-attachments/assets/732a4132-2857-49fc-97a4-6dfcea5364f8">
</p>
</details> 

I didn't find any example using Kotlin 2.0.20 and Koin Annotations using shared native modules on the internet.
Am I missing something?

**To Reproduce**
Here is a fork of InsertKoinIO/KMP-App-Template that reproduces this issue:
https://github.com/vinceglb/KMP-App-Template/tree/upgrade-to-koin-4.0

**Expected behavior**
Be able to use Koin 4.0, Koin Annotations and Kotlin 2 to declare shared common modules in a KMP project.

**Koin project used and used version (please complete the following information):**
```toml
koin = "4.0.0"
koin-annotations = "2.0.0-Beta1"
kotlin = "2.0.20"
ksp = "2.0.20-1.0.25"
```

**Comment by deniskniazev-municorn** [comment]
If I got your ticket right, I faced the same error. Koin generates module file for every target, f.e. `module/build/generated/ksp/iosSimulatorArm64/iosSimulatorArm64Main/kotlin/org/koin/ksp/generated/FeatureModuleGencom`. And I cannot use generated `org.koin.ksp.generated.module` in commonMain code with last lib versions.

As workaround I created in commonMain 
`expect val featureModule: org.koin.core.module.Module`
and for each target (f.e. in iosSimulatorArm64Main )
`actual val featureModule = FeatureModule().module`
